### ISSUE TITLE
Error npm install

### ISSUE THREAD

**Issue by TheIndra55** [cubehouse/themeparks#50]
Hello!

When i'm using the npm install command i get the following error :
`node-pre-gyp ERR! Tried to download(404): https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v11-linux-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for libxmljs@0.18.4 and node@0.10.25 (node-v11 ABI) (falling back to source compile with node-gyp)
node-pre-gyp http 404 status code downloading tarball https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v11-linux-x64.tar.gz`

After installing i get 
`
`Error:` Cannot find module 'themeparks'

**Comment by cubehouse** [comment]
Hi,

This is more an issue with libxmljs. What OS are you using? If you're on Ubuntu, can you try installing "build-essential" to see if that will build the library from source?

**Comment by TheIndra55** [comment]
@cubehouse Hi,
I'm using ubuntu 15.04 
I installed build-essentials, Is there any other way to get it to work?

**Comment by cubehouse** [comment]
Do you have all the software required for node-gyp? https://github.com/nodejs/node-gyp#installation

Make sure you can install libxmljs and follow any support for that. If you can let me know what works, I can make sure the README is correct so that users are aware of any basic steps they need to follow for libxmljs to work.

**Comment by TheIndra55** [comment]
Hi @cubehouse 
Because i couldn't get it to work i writed a php code (I only need disney wait times)
This code succesfull get's the acces_token and logs in at the api of disney

But i can't find the right url to get the wait times
I'm using this url right now
`https://api.wdpro.disney.go.com/facility-service/theme-parks/P1;location=dlp/wait-times`

I'm getting the following api output'
no resource for specified id was foundnull

Can you give me some info about the disneyland paris right url?

**Comment by cubehouse** [comment]
Hi,

I'm afraid I cannot give support for alternative implementations of this library, I only intend to support this codebase and have no interest in supporting implementations in other languages.

I have however reproduced the issue you had originally in a fresh Ubuntu install. I followed the error messages and while this is really an install issue with libxmljs, the node-gyp readme told me what I needed to install:
    
    apt-get install build-essential python2.7 -y

I will update the documentation later to make this clear.

**Comment by fyroc** [comment]
I am also having the same issue. I am trying to run this on CentOS. It seems a bit odd that there are extra server side installations needed for an API library.

Can you not fix the package itself rather than this work around?

**Comment by fyroc** [comment]
@TheIndra55 Shoot me an email.

**Comment by cubehouse** [comment]
As said before, this is because of another package, libxmljs.

We're not actually using that package directly, but we're using osmosis, which then uses libxmljs, which then uses node-gyp to compile if you're not on Windows (it supplies pretty compiled binaries for Windows only for some reason).
I'm looking into osmosis alternatives, but I don't think installing build tools is such an unusual step for server software.

I'll look into replacing the package dependency with an alternative, but that will require rewriting any part of that library using it, so it will not be soon.

**Comment by TheIndra55** [comment]
Did you already update the documentation?

**Comment by cubehouse** [comment]
It's on my todo list, but is low priority. I'll get round to it soon, but I need to investigate how this would apply to other distributions, and see how any other projects deal with this.

If you are waiting on documentation, everything you need is in this ticket. Installing make and python27 are the key components, if there is anything else, check the node-gyp documentation.

**Comment by TheIndra55** [comment]
Thanks! Everything works fine on my server now

**Comment by cubehouse** [comment]
4.3.0 is now live and no longer relies on osmosis, which in turn relied on libxml, which caused most pain-points for people trying to install this library.

I know we resolved this for most people already through helping people install libxml correctly, but I didn't want to resolve this ticket until the library was easier to install for all users, so no closing.
